AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -67,6 +94,29 @@ final class HiddenCoreDataMeasurementStreamStorage: MeasurementStreamStorageCont
         self.context = context
     }
     
+    func deleteSession(_ sessionUUID: SessionUUID) throws {
+        do {
+            try context.delete(context.existingSession(uuid: sessionUUID))
+        } catch {
+            Log.error("Error when deleting session")
+        }
+    }
+    
+    func deleteStreams(_ sessionUUID: SessionUUID) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        let toDelete = sessionEntity.allStreams!.filter({ $0.gotDeleted })
+        toDelete.forEach { object in
+            do {
+                toDelete.forEach { object in
+                    context.delete(object)
+                }
+            } catch {
+                Log.error("Error when deleting session")
+            }
+        }
+        sessionEntity.changesCount += 1
```

Comment:
We perhaps need the same comment here too (or maybe move this line to a separate function that will be invoked in both places and only comment this func?)